libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxSpeechApi.kt

```diff
@@ -73,12 +118,50 @@ class MapboxSpeechApi @JvmOverloads constructor(
      */
     fun clean(announcement: SpeechAnnouncement) {
         voiceAPI.clean(announcement)
+        VoiceInstructionsParser.parse(announcement).onValue {
+            synchronized(cachedFilesLock) {
+                val value = cachedFiles[it]
+                // when we clear fallback announcement, there is a chance we will remove the key
+                // from map and not remove the file itself
+                // since for fallback SpeechAnnouncement file is null
+                if (value?.announcement == announcement) {
+                    cachedFiles.remove(it)
+                }
+            }
+        }
+    }
+
+    internal fun predownload(instructions: List<VoiceInstructions>) {
+        mainJobController.scope.launch {
+            instructions.forEach { instruction ->
+                val typeAndAnnouncement = VoiceInstructionsParser.parse(instruction).value
+                if (typeAndAnnouncement != null && !hasTypeAndAnnouncement(typeAndAnnouncement)) {
+                    predownloadScope.launch {
+                        retrieveVoiceFile(instruction) {
+                            it.onValue { speechValue ->
+                                synchronized(cachedFilesLock) {
+                                    cachedFiles[typeAndAnnouncement] = speechValue
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    internal fun destroy() {
```

Comment:
do you want to support reusing of the same object after destroy?
If yes, then you probably need to recreate scope.
If no, then why do you need to cleanup `cachedFiles`?

Reply:
Yes, I've stumbled across this problem while writing tests. I've changed the behaviour to support reusing it.